src/row.rs

```diff
@@ -95,6 +96,16 @@ impl Row {
 		}
 	}
 
+    // You need to impl Iterator for these to work with for / map implicitly
+    // But there is a compiler warning when you do add the trait.

```

Comment:
To answer you, I'm not sure you can implement `Iterator` trait over an iterable struct. An iterator is used to iterate over an iterable. You have to explicitly call `iter_mut` if you want to call `map`